src/main/kotlin/ar/edu/unahur/obj2/socios/Cliente.kt

```diff
@@ -1,4 +1,8 @@
 package ar.edu.unahur.obj2.socios
 
-class Cliente {
-}
+class Cliente (var estadoDeAnimo : EstadoDeAnimo, var dineroEnBolsillo : Int, var barrio : Barrio) {
+    fun cuantaPropinaDejaria(importe : Int): Int {
+        val segunEstado = estadoDeAnimo.influirPropina(this,importe)
+        return segunEstado + barrio.influirPropina(this, segunEstado)
+    }
```

Comment:
Al valor dado del animo se le tenía que agregar o calcular lo del barrio. Una solución mejor sería.
```suggestion
    fun cuantaPropinaDejaria(importe : Int) = barrio.influirPropina(estadoDeAnimo.influirPropina(this,importe))
```